### ISSUE TITLE
TykMakeHttpRequest should obey proxy_ssl_insecure_skip_verify

### ISSUE THREAD

**Issue by bitsofinfo** [TykTechnologies/tyk#1006]
**Do you want to request a *feature* or report a *bug*?**

feature/bug

**What is the current behavior?**

`TykMakeHttpRequest` can't make calls to self-signed CA or mis-matched hostname backends

```
[JSVM]: Request failed: Get https://mybackend.mydomain.com/whatever: x509: certificate is valid for xxxxx, not mybackend.mydomain.com"
```

**What is the expected behavior?**

There should be an option/param/flag for `TykMakeHttpRequest` OR it should just obey what the gateway's `proxy_ssl_insecure_skip_verify` setting says

**Which versions of Tyk affected by this issue? Did this work in previous versions of Tyk?**

2.3.7

**Comment by lonelycode** [comment]
One workaround for this is to add the certificate to the trusted certificate store on the requesting box, it should bubble through.  

`TykMakeHttpRequest` does not use the proxy, so it will use a different client, which is why the setting isn't applied. Though I guess it should.

**Comment by bitsofinfo** [comment]
K, then I'd suggest another flag arg to the `TykMakeHttpRequest` component to toggle this

**Comment by lonelycode** [comment]
The workaround described above is this:

https://unix.stackexchange.com/questions/90450/adding-a-self-signed-certificate-to-the-trusted-list

**Comment by paltryeffort** [comment]
Doing a curl in the container without certificate I get:
curl: (60) SSL certificate problem: self signed certificate

After adding the certificate to the container I get:
curl: (51) SSL: certificate subject name 'something' does not match target host name 'myserver'

TykMakeHttpRequest doesn't seem to work with that. Is there any workaround for that?

**Comment by buger** [comment]
We already have `ProxySSLInsecureSkipVerify` flag in configuration, and probably it makes sense `TykMakeHttpRequest` to respect it.

**Comment by bitsofinfo** [comment]
Yes, however in the meantime, is there any other workaround to make this work?

**Comment by buger** [comment]
Make local /etc/hosts record with valid domain name specified in the certificate?

**Comment by bitsofinfo** [comment]
this isn't going to work, the backends are more containers behind a LB, and the names of things are dynamic. When can this support for `TykMakeHttpRequest ` to obey this be released in a patch?

**Comment by mvdan** [comment]
@bitsofinfo given that this is an enhancement, it doesn't automatically qualify for a bugfix (patch) release. We will try to include this with 2.4, which is the upcoming major release.

Of course, if that can't wait, you're free to work on a fix and try it out by building from source - it shouldn't be too complicated. And you could also open a pull request to upstream the fix back to this repo.

Hope this helps.

**Comment by bitsofinfo** [comment]
Any update on when this might make it in?

**Comment by buger** [comment]
@bitsofinfo will be addressed as part of 2.4.2, coming next week
